### PR TITLE
Add YOLO v3 model for object detection and classification

### PR DESCRIPTION
**Pull request by simgae**
Add a new YOLO v3 model for multiple object detection and classification in the `src/models` folder, implemented in Python with TensorFlow, trained on the KITTI dataset.

* **New YOLO v3 Model Implementation**
  - Add `src/models/yolo_v3_model.py` file.
  - Define the YOLO v3 model architecture in the `YoloV3Model` class.
  - Implement the `train_model` method to use the YOLO v3 model.
  - Implement the `evaluate_model` method to use the YOLO v3 model.
  - Implement the `_evaluate_image` method to use the YOLO v3 model.
  - Load and preprocess the KITTI dataset.
  - Compile the model with a custom YOLO loss function.
  - Train and evaluate the model using the KITTI dataset.

* **Update Initialization**
  - Modify `src/__init__.py` to import the `YoloV3Model` class from `src.models.yolo_v3_model`.
  - Update the instantiation of the YOLO model to use `YoloV3Model`.
  - Update the commented-out lines to use `YoloV3Model` methods.

### CODE REVIEW

File: src/models/yolo_v3_model.py [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+        Evaluates the YOLO v3 model's performance on the test dataset.
+        """
+        test_loss = self.model.evaluate(self.test_dataset)
+        print(f"Test Loss: {test_loss}")
+
+    def load_model(self):
+        """
+        Loads a trained YOLO v3 model from a file.
+        """
+        self.model = keras.models.load_model('yolo_v3_model.keras',
```
Comment (ischmid) [inline src/models/yolo_v3_model.py:187]: bitte den Pfad mit src/yolo_v3_model.keras angeben, sonst kann ich das nicht finden

File: src/models/yolo_v3_model.py [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+            tf.expand_dims(bbox, 0),
+            colors
+        )
+
+        image = tf.squeeze(image, 0)
+
+        encoded_image = tf.image.encode_png(tf.image.convert_image_dtype(image, dtype=tf.uint8))
+
+        # save image
+        tf.io.write_file('output.jpg', encoded_image)
```
Comment (ischmid) [inline src/models/yolo_v3_model.py:291]: könntest du den Pfad bitte in 'output_images/output_yolo_v3.jpg' ändern?

**Review by LeonB3006 — APPROVED** [review]
